Submitted By:            Armin K. <krejzi at email dot com>
Date:                    2012-08-22
Initial Package Version: 2.16.0
Upstream Status:         Not Applicable
Origin:                  Self
Description:             This patch removes references to /var/db directory which is not part
                         of FHS and replaces them with more suitable directories in /var
                         hierarchy.

--- glibc.orig/Makeconfig	2012-06-30 21:12:34.000000000 +0200
+++ glibc/Makeconfig	2012-08-22 16:55:41.529395336 +0200
@@ -287,7 +287,7 @@
 
 # Directory for the database files and Makefile for nss_db.
 ifndef vardbdir
-vardbdir = $(localstatedir)/db
+vardbdir = $(localstatedir)/lib/nss_db
 endif
 inst_vardbdir = $(install_root)$(vardbdir)
 
--- glibc.orig/nscd/nscd.h	2012-06-30 21:12:34.000000000 +0200
+++ glibc/nscd/nscd.h	2012-08-22 16:53:50.531888590 +0200
@@ -112,11 +112,11 @@
 
 
 /* Paths of the file for the persistent storage.  */
-#define _PATH_NSCD_PASSWD_DB	"/var/db/nscd/passwd"
-#define _PATH_NSCD_GROUP_DB	"/var/db/nscd/group"
-#define _PATH_NSCD_HOSTS_DB	"/var/db/nscd/hosts"
-#define _PATH_NSCD_SERVICES_DB	"/var/db/nscd/services"
-#define _PATH_NSCD_NETGROUP_DB	"/var/db/nscd/netgroup"
+#define _PATH_NSCD_PASSWD_DB	"/var/cache/nscd/passwd"
+#define _PATH_NSCD_GROUP_DB	"/var/cache/nscd/group"
+#define _PATH_NSCD_HOSTS_DB	"/var/cache/nscd/hosts"
+#define _PATH_NSCD_SERVICES_DB	"/var/cache/nscd/services"
+#define _PATH_NSCD_NETGROUP_DB	"/var/cache/nscd/netgroup"
 
 /* Path used when not using persistent storage.  */
 #define _PATH_NSCD_XYZ_DB_TMP	"/var/run/nscd/dbXXXXXX"
--- glibc.orig/nss/db-Makefile	2012-06-30 21:12:34.000000000 +0200
+++ glibc/nss/db-Makefile	2012-08-22 16:55:05.688908816 +0200
@@ -22,7 +22,7 @@
 		       /etc/rpc /etc/services /etc/shadow /etc/gshadow \
 		       /etc/netgroup)
 
-VAR_DB = /var/db
+VAR_DB = /var/lib/nss_db
 
 AWK = awk
 MAKEDB = makedb --quiet
--- glibc.orig/sysdeps/generic/paths.h	2012-06-30 21:12:34.000000000 +0200
+++ glibc/sysdeps/generic/paths.h	2012-08-22 16:56:19.705913564 +0200
@@ -68,7 +68,7 @@
 /* Provide trailing slash, since mostly used for building pathnames. */
 #define	_PATH_DEV	"/dev/"
 #define	_PATH_TMP	"/tmp/"
-#define	_PATH_VARDB	"/var/db/"
+#define	_PATH_VARDB	"/var/lib/nss_db/"
 #define	_PATH_VARRUN	"/var/run/"
 #define	_PATH_VARTMP	"/var/tmp/"
 
--- glibc.orig/sysdeps/unix/sysv/linux/paths.h	2012-06-30 21:12:34.000000000 +0200
+++ glibc/sysdeps/unix/sysv/linux/paths.h	2012-08-22 16:56:32.386085693 +0200
@@ -68,7 +68,7 @@
 /* Provide trailing slash, since mostly used for building pathnames. */
 #define	_PATH_DEV	"/dev/"
 #define	_PATH_TMP	"/tmp/"
-#define	_PATH_VARDB	"/var/db/"
+#define	_PATH_VARDB	"/var/lib/nss_db/"
 #define	_PATH_VARRUN	"/var/run/"
 #define	_PATH_VARTMP	"/var/tmp/"
 
